cdc/sinkv2/eventsink/mq/producer/producer.go

```diff
@@ -27,8 +28,9 @@ type Producer interface {
 	) error
 
 	// Close closes the producer and client(s).
-	Close() error
+	Close()
 }
 
-// NewProducerFunc is a function to create a producer.
-type NewProducerFunc func() Producer
+// Creator is a function to create a producer.
+type Creator func(ctx context.Context, client sarama.Client,
+	errCh chan error) (Producer, error)
```

Comment:
Please add comment on how `errCh` is supposed to be used.
- Who is responsible for closing it?
- What's the expectation for the buffer size?

Reply:
Added. Please checkout [6256b2e](https://github.com/pingcap/tiflow/pull/6412/commits/6256b2e50e3e3b0ff3fc1827448c7655d7e35216)